src/Executables/PrintMinimumGridSpacing/PrintMinimumGridSpacing.hpp

```diff
@@ -0,0 +1,258 @@
+          domain = std::move(*domain_ptr);
+        });
+
+    return std::make_tuple(
+        db::create_from<db::RemoveTags<>,
+                        db::AddSimpleTags<>,
+                        compute_tags>(
+            elliptic::Initialization::Domain<Dim>::initialize(
+                db::create_from<typename AddOptionsToDataBox::simple_tags>(
+                    std::move(box)),
+                array_index, initial_extents, domain)),
+        true);
+  }
+
+  template <typename DbTagsList, typename... InboxTags, typename Metavariables,
+            typename ActionList, typename ParallelComponent,
+            Requires<not tmpl::list_contains_v<DbTagsList, Domain>> = nullptr>
+  static std::tuple<db::DataBox<DbTagsList>&&, bool> apply(
+      db::DataBox<DbTagsList>& box,
+      const tuples::TaggedTuple<InboxTags...>& /*inboxes*/,
+      const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+      const ElementIndex<Dim>& /*array_index*/, const ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) noexcept {
+    return {std::move(box), true};
+  }
+};
+
+struct PrintMinimumGridSpacing {
+  template <typename ParallelComponent, typename DbTags, typename Metavariables,
+            typename ArrayIndex>
+  static void apply(db::DataBox<DbTags>& /*box*/,
+                    const Parallel::ConstGlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/,
+                    const double& overall_min_grid_spacing) noexcept {
+    Parallel::printf("Overall inertial minimum grid spacing: %f\n\n",
+              overall_min_grid_spacing);
+  }
+};
+
+template <size_t Dim>
+struct ArrayReduce {
+  template <typename ParallelComponent, typename DbTags, typename Metavariables,
+            typename ArrayIndex,
+            Requires<tmpl::list_contains_v<DbTags, Tags::Mesh<Dim>>> = nullptr>
+  static void apply(const db::DataBox<DbTags>& box,
+                    const Parallel::ConstGlobalCache<Metavariables>& cache,
+                    const ArrayIndex& array_index) noexcept {
+    static_assert(cpp17::is_same_v<ParallelComponent,
+                                   ElementArray<Dim, Metavariables>>,
+                  "The ParallelComponent is not deduced to be the right type");
+    const auto& my_proxy =
+        Parallel::get_parallel_component<ElementArray<Dim, Metavariables>>(
+            cache)[array_index];
+    const auto& singleton_proxy = Parallel::get_parallel_component<
+        SingletonParallelComponent<Dim, Metavariables>>(cache);
+
+    Parallel::ReductionData<Parallel::ReductionDatum<double, funcl::Min<>>>
+        elemental_min_grid_spacing{
+            get<Tags::MinimumGridSpacing<Dim, Frame::Inertial>>(box)};
+    Parallel::contribute_to_reduction<PrintMinimumGridSpacing>(
+        elemental_min_grid_spacing, my_proxy, singleton_proxy);
+  }
+};
+}  // namespace Actions
+
+template <size_t Dim, typename Metavariables>
+struct SingletonParallelComponent {
```

Comment:
I suggest to name this component something like `Printer`, to describe what it does rather than its type. (Please feel free to use a more imaginative name if you come up with one 😄)

Reply:
I don't find the name "Printer" very informative...